tests/unittests/Foundation/NSPointerArrayTests.mm

```diff
@@ -259,8 +243,14 @@ @implementation NSPointerArraySubClass
     ASSERT_EQ_MSG(array.count, 2, @"Incorrect size");
 }
 
-TEST(NSPointerArray, Weak) {
-    NSPointerArray* array = [NSPointerArray weakObjectsPointerArray];
+// TODO::
+// Bug #803:NSArrayPointer should implement dealloc observer for objects added to it
+// Disabling test until the bug is fixed.
+//
+DISABLED_TEST(NSPointerArray, Weak) {
+    // Create a autorelease pool for this test so that the objects' used by this test are managed locally.
+    NSAutoreleasePool* pool = [NSAutoreleasePool new];
```

Comment:
>NSAutoreleasePool* pool = [NSAutoreleasePool new]; [](start = 4, length = 50)

What happens if you move this one line lower, so that array is not managed by this autorelease pool? (It should probably be one line lower anyway, because the memory management scopes are currently criss-crossing)